### PR TITLE
Minor improvments & bug fixes

### PR DESCRIPTION
**Pull request by terziman**
Hi,

This pull request contains the few minor improvments and bug fixes I have made over the last year.

In particular, it contains an improved handling for transparency in collada. Indeed, half the exporters handles the transparency value in a wrong way. Therefore, the importer now have an option to select how to interpret the transparency value.

The prototype of the SetProperty method was also improved to avoid ugly boolean pointer passed as parameter instead of using a simple return value. In particular, this modification make those functions much more easier to wrap into another language.

Finally, some of the commits also fixes some compilation errors or warnings when using mingw.

I hope you will find this useful.

Sincerely

Leo Terziman

### CODE REVIEW

**Comment by LegalizeAdulthood** [comment]
FYI, Travis says the unit tests are failnig to compile because of the `bool` signature change.

**Comment by terziman** [comment]
I know. I've been working on the fix. I will submit another pull request.

**Comment by terziman** [comment]
By the way, the unit test does not support out-of-source build configurations with cmake: some references to "test/models/..." are hard coded.

File: code/Exporter.cpp [hunk 1]
```diff
@@ -513,34 +513,30 @@ ExportProperties::ExportProperties(const ExportProperties &other)
 
 // ------------------------------------------------------------------------------------------------
 // Set a configuration property
-void ExportProperties :: SetPropertyInteger(const char* szName, int iValue, 
-	bool* bWasExisting /*= NULL*/)
+bool ExportProperties :: SetPropertyInteger(const char* szName, int iValue)
 {
-	SetGenericProperty<int>(mIntProperties, szName,iValue,bWasExisting);
+	return SetGenericProperty<int>(mIntProperties, szName,iValue);
```
Comment (acgessler) [inline code/Exporter.cpp]: Agree.

File: code/Importer.cpp [hunk 2]
```diff
@@ -925,42 +925,46 @@ void Importer::GetExtensionList(aiString& szOut) const
 
 // ------------------------------------------------------------------------------------------------
 // Set a configuration property
-void Importer::SetPropertyInteger(const char* szName, int iValue, 
-	bool* bWasExisting /*= NULL*/)
+bool Importer::SetPropertyInteger(const char* szName, int iValue)
 {
+	bool exising;
```
Comment (acgessler) [inline code/Importer.cpp]: Please rename `existing`.

File: include/assimp/ai_assert.h [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 #ifdef ASSIMP_BUILD_DEBUG  
 #	include <assert.h>
-#	define	ai_assert(expression) assert(expression)
+#	define	ai_assert(expression) assert(expression);
```
Comment (acgessler) [inline include/assimp/ai_assert.h]: Please undo this change, this is counter-intuitive behaviour for a macro.

**Comment by acgessler** [comment]
Thanks!
